### PR TITLE
Launchpad: Add Dismiss launchpad option

### LINKED ISSUE #85037: Launchpad: Add the possibility of dismissing the Launchpad without completing all the tasks
Discussing with @daledupreez, we talked about the idea of making it possible to dismiss the Launchpad on the Customer's Home without completing all the tasks. But it can also be used in other contexts too.

- [x] Add a flag to the task list definition to allow the user to dismiss the Launchpad without completing all the tasks
- [x] Make the necessary changes on the front end to accept the new flag

We should also explore how we are going to display the "Dismiss" button alongside the progress ring. We may need @nuriapenya to chime on this.

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #85557
Depends on https://github.com/Automattic/jetpack/pull/34889

## Proposed Changes
* Introduce the Dismiss settings menu only when the launchpad is dismissible
This PR introduces the concept of temporary and permanent launchpad dismiss. Temporary dismiss is based on time, so the frontend is setting a time in the future to describe when the launchpad will be available again. 
The permanent dismiss is based on the already existent 'is_dimissed' flag. 

image

## Testing Instructions
* Set your sandbox with the PR https://github.com/Automattic/jetpack/pull/34889
* Set the build tasklist as dismissible, similar to this change https://github.com/Automattic/jetpack/pull/34839/files#diff-7a7ab71e45292d18195f8de5e1f0338aa5232291425fa0bbe9a21afa65bca23fR273
* Load the site home home/<YOUR_SITE>
* Check if the 3 dots button is rendered
* Choose one of the temporary options following the tip above 
* Check the last option (forever) 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by valterlorran — CHANGES_REQUESTED** [review]
Well done! Big kudos for the tests!

We'll need to fix the arguments related to the changes you did on the site slug types, you can check that on the failing actions. Perhaps we should tackle those changes in another PR.

**Comment by gabrielcaires** [comment]
> Well done! Big kudos for the tests!
> 
> We'll need to fix the arguments related to the changes you did on the site slug types, you can check that on the failing actions. Perhaps we should tackle those changes in another PR.

Types fixed https://github.com/Automattic/wp-calypso/pull/86010/commits/9a37dbf89b1f8ccd89d140146ac506f3f85d446a as we discussed.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10746148

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @gabrielcaires for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/customer-home/cards/launchpad/index.tsx [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
 
 	// return nothing if the launchpad is dismissed
 	if ( isDismissed ) {
-		return <></>;
+		return null;
 	}
 
+	const temporaryDismiss = ( duration: Duration ) => {
+		dismiss( {
+			dismissedUntil: moment().add( duration ).utc().unix(),
```
Comment (valterlorran) [inline client/my-sites/customer-home/cards/launchpad/index.tsx]: I believe we shouldn't be sending the time from the front end. We should be sending `day,`/`week,` and we would do the conversion to the Unix time on the back.

Comment (gabrielcaires) [reply client/my-sites/customer-home/cards/launchpad/index.tsx]: Do we have some specific reason for it? 

I think this approach enable us to have more variations with no backend changes, like 2 hours,  8 hours, etc.....

Comment (valterlorran) [reply client/my-sites/customer-home/cards/launchpad/index.tsx]: Trusting dates coming from the front end can lead to some issues. Even though you made sure we are using UTC, it assumes that the client's time is correct, which may not be true. Another problem I see is the possibility of inputting any other number maliciously, and we'd still calculate it and store it. In this case, it is completely harmless code that takes care of the back end, but we should still prevent such things from happening.

So, I would rather have a string sent to the back with the representation of the time frame, and we then get it translated and validated on the back end.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10746148).

File: client/my-sites/customer-home/cards/launchpad/index.tsx (hunk 1 above)
Comment (gabrielcaires) [reply client/my-sites/customer-home/cards/launchpad/index.tsx]: @valterlorran From security perspective, I think both solutions relays to the same problem, we need to validate the input, regardless the date format (string or number). I am going to add validations on the backend, regardless the solution.

I am going to change to relative time, thanks for the clarification.

Comment (gabrielcaires) [reply client/my-sites/customer-home/cards/launchpad/index.tsx]: @valterlorran I updated the frontend and the backend to use relative values. 
* Frontend [4f60c78](https://github.com/Automattic/wp-calypso/pull/86010/commits/4f60c780fa03e03d2c9bbca470f4283e9fb4a7c3)

* Backend https://github.com/Automattic/jetpack/pull/34889/commits/0366b10171fc4ea0683b6f4cf1171a97c52441ac

**Review by valterlorran — APPROVED** [review]
LGTM! Thanks for tackling all the issues!

**Pull request merged**